src/main/java/com/amazon/opendistroforelasticsearch/sql/query/DefaultQueryAction.java

```diff
@@ -208,9 +209,23 @@ private void setWhere(Where where) throws SqlParseException {
     private void setSorts(List<Order> orderBys) {
         for (Order order : orderBys) {
             if (order.getNestedPath() != null) {
-                request.addSort(SortBuilders.fieldSort(order.getName()).order(SortOrder.valueOf(order.getType())).setNestedSort(new NestedSortBuilder(order.getNestedPath())));
+                request.addSort(
+                        SortBuilders.fieldSort(order.getName())
+                                .order(SortOrder.valueOf(order.getType()))
+                                .setNestedSort(new NestedSortBuilder(order.getNestedPath())));
             } else {
-                request.addSort(order.getName(), SortOrder.valueOf(order.getType()));
+                if (order.isScriptField()) {
+                    // TODO: Investigate how to find the type of expression (string or number)
+                    // As of now this shouldn't be a problem, because the support is for date_format function
+
+                    ScriptSortBuilder.ScriptSortType string = ScriptSortBuilder.ScriptSortType.STRING;
+                    request.addSort(
+                            SortBuilders
+                                    .scriptSort(new Script(order.getName()), string)
```

Comment:
Usage of `string` as an identifier is confusing, plus, I think it would be more readable to just use the enum value inline, instead of assigning it to a local var.